tests/system_tests_tcp_adaptor_tls.py

```diff
@@ -30,15 +30,101 @@ def test_authenticate_peer(self):
             self.skipTest("Ncat utility is not available")
         name = "test_authenticate_peer"
         self.logger.log("TCP_TEST TLS Start %s" % name)
-        # Now, run ncat with a client cert and this time it should pass.
-        self.ncat_runner(name, client="INTA",
+        # Now, run ncat with a client cert and it should pass.
+        self.ncat_runner(name,
+                         client="INTA",
                          server="INTA",
                          logger=self.logger,
                          ncat_port=self.authenticate_peer_port,
                          use_ssl=True,
                          use_client_cert=True)
         self.logger.log("TCP_TEST Stop %s SUCCESS" % name)
 
+    def test_authenticate_peer_fail(self):
+        if not ncat_available():
+            self.skipTest("Ncat utility is not available")
+        name = "test_authenticate_peer_fail"
+        self.logger.log("TCP_TEST TLS Start %s" % name)
+        # Now, run ncat without a client cert and it should fail.
+        self.ncat_runner(name,
+                         client="INTA",
+                         server="INTA",
+                         logger=self.logger,
+                         ncat_port=self.authenticate_peer_port,
+                         use_ssl=True,
+                         use_client_cert=False,
+                         error_ok=True)
+        # Look for a log line that proves the peer did not return a certificate.
+        self.INTA.wait_log_message("peer did not return a certificate")
+        self.logger.log("TCP_TEST Stop %s SUCCESS" % name)
+
+    def test_connector_bad_server_cert(self):
+        if not ncat_available():
+            self.skipTest("Ncat utility is not available")
+        name = "test_connector_bad_server_cert"
+        self.logger.log("TCP_TEST TLS Start %s" % name)
+
+        # The router presents a cert to the echo server but the echo server is
+        # responding back with a different ca cert that the router cannot verify.
+        # Hence the router side TLS fails certificate verification.
+        self.ncat_runner(name,
+                         client="INTA",
+                         server="INTA",
+                         logger=self.logger,
+                         ncat_port=self.bad_server_port,
+                         use_ssl=True,
+                         use_client_cert=False,
+                         error_ok=True)
+        # Look for a log line that proves the certificate verification failure.
+        self.INTA.wait_log_message("certificate verify failed")
+        self.logger.log("TCP_TEST Stop %s SUCCESS" % name)
+
+    def test_xxx_present_bad_cert_to_good_listener_good_connector(self):
+        if not ncat_available():
+            self.skipTest("Ncat utility is not available")
```

Comment:
This could be something like
```suggestion
    @unittests.skipIf(not ncat_available(), "Ncat utility is not available")
    def test_xxx_present_bad_cert_to_good_listener_good_connector(self):
```